### PR TITLE
Fixed c++ compiler build warnings on prog28 and prog29

### PR DESCRIPTION
**Pull request by AgoinsUSGS**
Changes include:  Catching IException as a reference rather than value, 
                              Not using Qt::ControlModifier as a boolean reference, 
                              Removed deprecated dynamic throw() call.

There are special notes for the fixes for the Qt::ControlModifier and throw() warnings in the PR.

### CODE REVIEW

File: isis/src/qisis/objs/CubeViewport/CubeViewport.cpp [hunk 1]
```diff
@@ -1776,7 +1776,7 @@ namespace Isis {
       e->accept();
     }
     else if ((e->key() == Qt::Key_C) &&
-             QApplication::keyboardModifiers() &&
+             QApplication::keyboardModifiers() &
              Qt::ControlModifier) {
```
Comment (AgoinsUSGS) [inline isis/src/qisis/objs/CubeViewport/CubeViewport.cpp:1780]: warning: enum constant in boolean context [-Wint-in-bool-context] was being thrown by using the logical &&, I replaced it with the bitwise & (only difference as I understand it is that && can short-circuit and & will not, which I don't see is a huge difference in this context). If someone see's an issue with this fix let me know

File: isis/src/mro/apps/hijitreg/Instrument.cpp [hunk 2]
```diff
@@ -414,7 +415,7 @@ namespace UA {
       return
         (int)rint(CCD_FOCAL_PLANE_X_OFFSETS_MM[CPMM]
                   / (CCD_PIXEL_SIZE_MM * binning));
-    }
```
Comment (AgoinsUSGS) [inline isis/src/mro/apps/hijitreg/Instrument.cpp:417]: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]

For this, we were using a dynamic throw block that would throw either of the 2 exceptions in the block itself. 
These dynamic throws are deprecated in c++11 and removed from c++17 onward, so I moved the individual throws to be outside the throw() block and got rid of it. If anyone see's an issue with this please let me know.

Comment (SgStapleton) [reply isis/src/mro/apps/hijitreg/Instrument.cpp:417]: Confirmed that this should not be an issue.

File: isis/src/qisis/objs/CubeViewport/CubeViewport.cpp (hunk 1 above)
Comment (SgStapleton) [reply isis/src/qisis/objs/CubeViewport/CubeViewport.cpp:1780]: Confirmed that functionality has not changed. Holding cntrl-c copies the filename to the clipboard in both the nightly build as well as this version.

**Comment by SgStapleton** [comment]
Everything looks good to go!

**Pull request merged**